chemicalx/pipeline.py

```diff
@@ -152,10 +184,12 @@ def pipeline(
     for _epoch in trange(epochs):
         for batch in train_generator:
             optimizer.zero_grad()
-            prediction = model(*model.unpack(batch))
-            loss_value = loss(prediction, batch.labels)
+
+            device_batch = to_device(model.unpack(batch), device)
```

Comment:
I think the batch generator should know what device to put the batches on so this doesn't have to be changed in the pipeline

Reply:
I.e., the `BatchGenerator.__init__` should take an optional `torch.device` (if not given, assume CPU) and the generation steps should take care of moving the tensors over to the appropriate device